basis for digital ocean 1-click template

kennyfuto opened the pull request:
basis for digital ocean 1-click template

bo0tzz commented on digital-ocean/files/var/lib/cloud/scripts/per-instance/001_onboot:
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+#Generate Mysql root password.
+root_mysql_pass=$(openssl rand -hex 24)
+debian_sys_maint_mysql_pass=$(openssl rand -hex 24)
+
+# Save the passwords
+cat > /root/.digitalocean_password <<EOM
+root_mysql_pass="${root_mysql_pass}"
+EOM
+
+# Configure MySQL root password
+mysqladmin -u root -h localhost password ${root_mysql_pass}
+
+mysql -uroot -p${root_mysql_pass} \
+      -e "ALTER USER 'debian-sys-maint'@'localhost' IDENTIFIED BY '${debian_sys_maint_mysql_pass}'"
+
+cat > /etc/mysql/debian.cnf <<EOM
+# Automatically generated for Debian scripts. DO NOT TOUCH!
+[client]
+host     = localhost
+user     = debian-sys-maint
+password = ${debian_sys_maint_mysql_pass}
+socket   = /var/run/mysqld/mysqld.sock
+[mysql_upgrade]
+host     = localhost
+user     = debian-sys-maint
+password = ${debian_sys_maint_mysql_pass}
+socket   = /var/run/mysqld/mysqld.sock
+EOM
```
Is this a remnant from the base template or is mysql actually used somewhere?

bo0tzz commented on digital-ocean/scripts/011-docker-compose.sh:
We can get this from the apt repo just like docker core right? https://docs.docker.com/compose/install/linux/#install-using-the-repository.

kennyfuto replied:
I matched docker's set up for the digital ocean template.  How it is defined here is how the 1-click docker image does it.

010-docker.sh installs from the https://download.docker.com/linux/ubuntu instead of the ubuntu 20.04 repo.

011-docker-compose.sh Is using the github install of the latest version instead of the OS.  I believe this is to enforce better compability to the  https://download.docker.com/linux/ubuntu repository, instead the ubuntu 20.04 one.

kennyfuto commented:
I added the newline characters where I can.

I removed the mysql digital ocean related config in the startup script.

I tested this with immich version v1.139.4
